### ISSUE TITLE
Clean up example directories

### ISSUE THREAD

**Issue by KeithWoods** [esp/esp-js#18]
* lift src/examples to /examples
* delete dist/examples so the dist folder is purely the files we need to distribute
* re-configure the build so the ES6 examples not in /examples get transpiled to a new bin folder, perhaps /examples/bin
* each dir in root/examples should be a standalone app with its own packages.json + 'npm start' command

**Comment by KeithWoods** [comment]
I've done the items mentioned above with some minor changes. 

> e-configure the build so the ES6 examples not in /examples get transpiled to a new bin folder, perhaps /examples/bin

For this I just used babel's cli tool to transpile and run node on the fly. Most of the examples were simple scripts that run via node so this is perfect. Note that the packages.json use babel-node that is installed at the top level, i.e. each example doesn't pull down babel locally.

> each dir in root/examples should be a standalone app with its own packages.json + 'npm start' command

For this I included the packages.json however it only contains the start script. It doesn't have a 'dependencies' list that sucks in esp-js from npm, rather the scripts still relatively reference esp.js in the dist folder. I think this makes sense as you want all the examples to be running against the local source.
